src/components/molecules/NavSearchBar/NavSearchBar.tsx

```diff
@@ -1,69 +1,76 @@
-import React, { useCallback, useEffect, useState } from "react";
+import React, { useCallback, useEffect, useState, ChangeEvent } from "react";
+import classNames from "classnames";
+
+import { faSearch } from "@fortawesome/free-solid-svg-icons";
 
 import { VenueEvent } from "types/venues";
 import { Room, RoomTypes } from "types/rooms";
 import { User } from "types/User";
 
 import { WithId } from "utils/id";
-import { currentVenueSelectorData, venueEventsSelector } from "utils/selectors";
 import { isTruthy } from "utils/types";
+import { formatToWeekdayFullname, formatUtcSeconds } from "utils/time";
+import { currentVenueSelectorData, venueEventsSelector } from "utils/selectors";
 
 import { useWorldUsers } from "hooks/users";
 import { useSelector } from "hooks/useSelector";
 import { useProfileModalControls } from "hooks/useProfileModalControls";
 
+import { InputField } from "components/atoms/InputField";
 import { RoomModal } from "components/templates/PartyMap/components";
 
+import { NavSearchResult } from "./NavSearchResult";
+
 import "./NavSearchBar.scss";
-import { InputField } from "components/atoms/InputField";
-import { faSearch } from "@fortawesome/free-solid-svg-icons";
 
-interface SearchResult {
-  rooms: Room[];
-  users: readonly WithId<User>[];
-  events: VenueEvent[];
-}
+type WithImageUrl<T extends object> = T & { image_url?: string };
+
+const emptyEventsArray: VenueEvent[] = [];
```

Comment:
I'm guessing we use this so that we have a consistent reference to the array when we use it as a default, rather than creating a new one on every render?

Reply:
yes, right, that was my idea.

Reply:
_nods_ Makes sense.